src/daemon/inout_route.rs

```diff
@@ -103,3 +101,66 @@ pub async fn out_route_obfsudp(
         .await;
     }
 }
+
+async fn per_route_tasks(
+    ctx: DaemonContext,
+    pipe: impl Pipe,
+    their_fp: Option<Fingerprint>,
+    link_price: LinkPrice,
+) -> anyhow::Result<()> {
+    let link_info =
+        LinkConnection::connect(ctx.clone(), pipe, link_price.max_outgoing_price).await?;
```

Comment:
the problem here is that LinkConnection is *still* a "facade" struct that communicates via channels to a task. moving the task out, through LinkInfo,  does not make things much cleaner.

instead, the LinkConnection struct should be removed (see other comments)